hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java

```diff
@@ -352,6 +355,7 @@ public Collection<V> values(Predicate predicate) {
     @Override
     public Set<Map.Entry<K, V>> entrySet(Predicate predicate) {
         checkNotNull(predicate, "Predicate cannot be null!");
+        PredicateUtils.checkDoesNotContainPagingPredicate(predicate, "entrySet");
```

Comment:
why do we have this check? shouldn't this method be able to specify a paging predicate?